clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -432,6 +432,24 @@ public ConfigDef withClientSaslSupport() {
     }
 
     /**
+     * Filter configs that are not defined in this configuration definition.
+     *
+     * @param props The configs to be filtered.
+     * @return Filtered configs.
+     */
+    public Map<String, Object> filter(Map<String, Object> props) {
```

Comment:
I thought about that, and was a bit worried about this approach since we have currently not exposed the config keys (they are considered private).

Let me see if there is a better way of exposing it.